ethereum/contracts/SimplifiedMMRVerification.sol

```diff
@@ -0,0 +1,96 @@
+// "SPDX-License-Identifier: UNLICENSED"
+pragma solidity ^0.8.5;
+
+struct SimplifiedMMRProof {
+    bytes32[] restOfThePeaks;
+    bytes32 rightBaggedPeak;
+    bytes32[] merkleProofItems;
+    uint64 merkleProofOrderBitField;
+}
+
+contract  SimplifiedMMRVerification {
+    function verifyInclusionProof(
+        bytes32 root,
+        bytes32 leafNodeHash,
+        SimplifiedMMRProof memory proof
+    ) public pure returns (bool) {
+        require(proof.merkleProofItems.length < 64);
+
+        bool hasRightBaggedPeak = proof.rightBaggedPeak != 0x0;
```

Comment:
@philipstanislaus :) That was the original design. @musnit asked me to change it this way to save on gas costs.

Reply:
@musnit what are your thoughts here, are we fine with a small likelihood that this will fail? It's probably not relevant in practice, but may shed a negative light on us in audits or completely break the protocol (same probability that I will guess your private key, and same probability that someone actually owns the private key for the 0x0 address on Ethereum).

Reply:
@philipstanislaus The peak is hash, so technically they would have to find a node which hashes to 0x0. This node has following structure: 
```
struct BeefyMMRLeafPartial {
        uint8 version;
        uint32 parentNumber;
        bytes32 parentHash;
        uint64 nextAuthoritySetId;
        uint32 nextAuthoritySetLen;
        bytes32 nextAuthoritySetRoot;
}
```

So, you will have to generate a node with all of this field valid and then that node need to hash 0x0 in order to do this.
I would consider probability of this extremely low.

Reply:
i don't think we need to be secure against someone finding the preimage of 0x0 - even with a [hypercomputer](https://en.wikipedia.org/wiki/Hypercomputation), that preimage is likely not even valid input given the constraints @ParthDesai mentions above :P

Reply:
Agree, but I’m less concerned about an attack, but rather about a failure if the actual/valid hash of the right bag is 0x0. In that case, a valid MMR proof will fail. 

But it’s not a real concern, because we could simply wait for the next beefy commitment, which changes the right bagged peaks hash, resolving the issue.

All good to keep it 👌👍